ecommerce/models.py

```diff
@@ -223,6 +228,44 @@ class Coupon(Model):
     )
     enabled = BooleanField(default=True, help_text="If true, coupons are presently redeemable")
 
+    @property
+    def course_keys(self):
+        """Get the course keys which the coupon can be redeemed with"""
+        obj = self.content_object
+        if isinstance(obj, Program):
+            return CourseRun.objects.filter(course__program=obj).values_list('edx_course_key', flat=True)
+        elif isinstance(obj, Course):
+            return CourseRun.objects.filter(course=obj).values_list('edx_course_key', flat=True)
+        else:
+            # This must be a CourseRun. This is validated in clean()
+            return [obj.edx_course_key]
+
+    @property
+    def is_valid(self):
+        """Returns true if the coupon is enabled and also within the valid date range"""
+        now = datetime.now(tz=pytz.UTC)
+        if not self.enabled:
+            return False
+        if self.activation_date is not None and now < self.activation_date:
+            return False
+        if self.expiration_date is not None and now > self.expiration_date:
+            return False
+        return True
+
+    @property
+    def is_automatic(self):
+        """
+        Returns true if the coupon would be redeemed automatically without input from the user.
+        """
+        return self.coupon_type == self.DISCOUNTED_PREVIOUS_COURSE
```

Comment:
Because there may be other coupons which are automatically applied but not necessarily due to having purchased the previous course. Keeping `is_automatic` separate from `DISCOUNTED_PREVIOUS_COURSE` means `is_automatic` can change to handle other types of coupons in the future